bitcoin/src/consensus/encode.rs

```diff
@@ -720,6 +720,7 @@ impl<T: Encodable> Encodable for rc::Rc<T> {
     }
 }
 
+#[cfg(feature = "std")]
```

Comment:
This is imprecise, there are `alloc` targets that don't need `std` for atomics.

[`#[cfg(target_has_atomic = "ptr")]` needs to be used instead](https://doc.rust-lang.org/stable/alloc/sync/index.html)

Also needs `docsrs` gate to document it - see our other gates.